quickwit/quickwit-metastore/src/metastore/postgres/utils.rs

```diff
@@ -179,6 +179,39 @@ pub(super) fn append_query_filters(sql: &mut SelectStatement, query: &ListSplits
         Expr::expr(val)
     });
 
+    match query.sort_by {
+        SortBy::Staleness => {
+            sql.order_by(
+                (Splits::DeleteOpstamp, Splits::PublishTimestamp),
+                Order::Asc,
+            );
+        }
+        SortBy::IndexUid => {
+            // this order by can be fairly costly,
+            // from testing, adding a join here was way faster, because we do an index-only scan on
+            // indexes.index_uid, nested-loop merged with a bitmap index scan on splits.index_uid,
+            // filter for our conditions, and just take the first N results. This is guaranteed to
+            // return correct result because indexes.index_uid is a non-null foreign key
+            //
+            // We also need to do .column((Splits::Table, Asterisk)) from the caller side, to not
+            // return unexpected columns
+            //
+            // On the other hand, without join, we do a seq scan on splits, sort everything, and
+            // truncate.
+            //
+            // Or we could just add a btree index to splits.index_uid. That might be the better
+            // long term solution.
+            sql.join(
+                JoinType::Join,
+                Indexes::Table,
+                Expr::col((Splits::Table, Splits::IndexUid))
+                    .equals((Indexes::Table, Indexes::IndexUid)),
+            )
```

Comment:
note: i think this is a bad idea to do this *long term*, but i'd rather this fix be made without a migration for now, so we can easily revert clusters affected to previous versions if we find things are somehow still not good. If you disagree, we can either create the required index manually, or make a migration